op-alt-da/daclient.go

```diff
@@ -55,6 +73,14 @@ func (c *DAClient) GetInput(ctx context.Context, comm CommitmentData) ([]byte, e
 	if resp.StatusCode == http.StatusNotFound {
 		return nil, ErrNotFound
 	}
+	if resp.StatusCode == http.StatusTeapot {
+		defer resp.Body.Close()
+		var invalidCommitmentReason [250]byte
+		// We discard the error as it only contains the reason for invalidity.
+		// We might end up with a partial reason, but the commitment should still be skipped.
+		_, _ = io.ReadFull(resp.Body, invalidCommitmentReason[:])
+		return nil, InvalidCommitmentError{Reason: string(invalidCommitmentReason[:])}
```

Comment:
Would it make sense to capture the exact number of bytes read? Something like this:

```
		buf := make([]byte, 250)
		n, _ := io.ReadFull(resp.Body, buf)
		return nil, InvalidCommitmentError{Reason: string(buf[:n])}
```

Reply:
Yes good catch! Changed code to use an `http.MaxBytesReader` instead: [9251a34](https://github.com/Layr-Labs/optimism/pull/45/commits/9251a34831a789525f5e4c726ed63e5a81ee19d6)